core/src/main/java/net/opentsdb/query/execution/serdes/JsonV3QuerySerdes.java

```diff
@@ -241,23 +254,30 @@ public Object call(final ArrayList<TimeSeriesStringId> ids)
 
             final List<String> sets =
                 Lists.newArrayListWithExpectedSize(result.timeSeries().size());
-            pairs.stream().parallel().forEach((pair) -> {
-              try {
-                serializeSeries(opts,
-                    pair.getValue(),
-                    ids != null ? ids.get(pair.getKey()) :
-                      (TimeSeriesStringId) pair.getValue().id(),
-                    json,
-                    null,
-                    spec_start,
-                    spec_end,
-                    result);
-              } catch (Exception e) {
-                LOG.error("Failed to serialize ts: " + series, e);
-                throw new QueryExecutionException("Unexpected exception "
-                    + "serializing ts: " + series, 0, e);
-              }
-            });
+            customThreadPool.submit(() -> 
+              pairs.stream().parallel().forEach((pair) -> {
+                try {
+                  serializeSeries(opts,
+                      pair.getValue(),
+                      ids != null ? ids.get(pair.getKey()) :
+                        (TimeSeriesStringId) pair.getValue().id(),
+                      json,
+                      null,
+                      spec_start,
+                      spec_end,
+                      result);
+                } catch (Exception e) {
+                  LOG.error("Failed to serialize ts: " + series, e);
+                  throw new QueryExecutionException("Unexpected exception "
+                      + "serializing ts: " + series, 0, e);
+                }
+                if (pair.getValue().types().contains(StatusType.TYPE)
+                    && pair.getValue().id() instanceof BaseTimeSeriesByteId) {
```

Comment:
won't happen that it's a byte ID series, always string.
BUT we need to sync on the namespace list.